From 8b510e8e7acc473528f55fcce04470049202b0c8 Mon Sep 17 00:00:00 2001
From: Yufen Yu <yuyufen@huawei.com>
Date: Mon, 4 Mar 2019 20:07:18 +0200
Subject: [PATCH 2/2] block: add a new flag BLK_MQ_POLL_CLASSIC for hybrid poll

For q->poll_nsec == -1, means doing classic poll, not hybrid poll.
We introduce a new flag BLK_MQ_POLL_CLASSIC to replace -1, which
may make code much easier to read.

Signed-off-by: Yufen Yu <yuyufen@huawei.com>
---
 block/blk-mq.c         | 4 +++-
 block/blk-sysfs.c      | 4 ++--
 include/linux/blkdev.h | 3 +++
 3 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index c4effb956ebc..70584933e80a 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -2872,7 +2872,7 @@ struct request_queue *blk_mq_init_allocated_queue(struct blk_mq_tag_set *set,
 	/*
 	 * Default to classic polling
 	 */
-	q->poll_nsec = -1;
+	q->poll_nsec = BLK_MQ_POLL_CLASSIC;
 
 	blk_mq_init_cpu_queues(q, set->nr_hw_queues);
 	blk_mq_add_queue_tag_set(set, q);
@@ -3360,6 +3360,8 @@ static bool blk_mq_poll_hybrid_sleep(struct request_queue *q,
 	 *  0:	use half of prev avg
 	 * >0:	use this specific value
 	 */
+	if (q->poll_nsec == BLK_MQ_POLL_CLASSIC)
+		return false;
 	if (q->poll_nsec > 0)
 		nsecs = q->poll_nsec;
 	else
diff --git a/block/blk-sysfs.c b/block/blk-sysfs.c
index 590d1ef2f961..8639e135687e 100644
--- a/block/blk-sysfs.c
+++ b/block/blk-sysfs.c
@@ -360,7 +360,7 @@ static ssize_t queue_poll_delay_show(struct request_queue *q, char *page)
 {
 	int val;
 
-	if (q->poll_nsec == -1)
+	if (q->poll_nsec == BLK_MQ_POLL_CLASSIC)
 		val = -1;
 	else
 		val = q->poll_nsec / 1000;
@@ -381,7 +381,7 @@ static ssize_t queue_poll_delay_store(struct request_queue *q, const char *page,
 		return err;
 
 	if (val == -1)
-		q->poll_nsec = -1;
+		q->poll_nsec = BLK_MQ_POLL_CLASSIC;
 	else
 		q->poll_nsec = val * 1000;
 
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 56d8c5a25ab5..347b121b0539 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -50,6 +50,9 @@ struct blk_stat_callback;
 /* Must be consistent with blk_mq_poll_stats_bkt() */
 #define BLK_MQ_POLL_STATS_BKTS 16
 
+/* Doing classic polling */
+#define BLK_MQ_POLL_CLASSIC -1
+
 /*
  * Maximum number of blkcg policies allowed to be registered concurrently.
  * Defined here to simplify include dependency.
-- 
2.21.0

